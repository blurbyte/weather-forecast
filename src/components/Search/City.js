import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import ListItem from './ListItem';
import CountryCode from './CountryCode';

const Caption = styled.p`
  padding: 0 1.2rem;
  margin: 0;
`;

const City = ({ name, country }) => (
  <ListItem>
    <Caption>
      {name} <CountryCode>{country}</CountryCode>
    </Caption>
  </ListItem>
);

City.propTypes = {
  name: PropTypes.string.isRequired,
  country: PropTypes.string.isRequired
};

export default City;
