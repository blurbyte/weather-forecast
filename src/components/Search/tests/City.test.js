import React from 'react';
import renderer from 'react-test-renderer';

import City from '../City';

test('renders correctly', () => {
  const props = {
    cityId: 234567,
    name: 'London',
    country: 'US'
  };

  const component = renderer.create(<City {...props} />);
  expect(component.toJSON()).toMatchSnapshot();
});
