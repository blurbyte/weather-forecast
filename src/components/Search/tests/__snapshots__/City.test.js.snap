// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li
  className="sc-bdVaJa erDMoc"
>
  <p
    className="sc-htpNat bhyEbC"
  >
    London
     
    <span
      className="sc-bwzfXH bqulyh"
    >
      US
    </span>
  </p>
</li>
`;
